flixel/tweens/FlxTween.hx

```diff
@@ -621,7 +745,120 @@ typedef TweenOptions = {
 	?onUpdate:TweenCallback,
 	?onComplete:TweenCallback,
 	?startDelay:Null<Float>,
-	?loopDelay:Null<Float>,
+	?loopDelay:Null<Float>
+}
+
+typedef VarTweenParams = {
+	Object:Dynamic,
+	Values:Dynamic,
+	?Duration:Float,
+	?Options:TweenOptions
+}
+
+typedef NumTweenParams = {
+	FromValue:Float,
+	ToValue:Float,
+	Duration:Float,
+	?Options:TweenOptions,
+	?TweenFunction:Float->Void
+}
+
+typedef AngleTweenParams = {
+	?Sprite:FlxSprite,
+	FromAngle:Float,
+	ToAngle:Float,
+	?Duration:Float,
+	?Options:TweenOptions
+}
+
+typedef ColorTweenParams = {
+	?Sprite:FlxSprite,
+	?Duration:Float,
+	FromColor:FlxColor,
+	ToColor:FlxColor,
+	?Options:TweenOptions
+}
+
+typedef LinearMotionParams = {
+	Object:FlxObject,
+	FromX:Float,
+	FromY:Float,
+	ToX:Float,
+	ToY:Float,
+	?DurationOrSpeed:Float,
+	?UseDuration:Bool,
+	?Options:TweenOptions
+}
+
+typedef QuadMotionParams = {
+	Object:FlxObject,
+	FromX:Float,
+	FromY:Float,
+	ControlX:Float,
+	ControlY:Float,
+	ToX:Float,
+	ToY:Float,
+	?DurationOrSpeed:Float,
+	UseDuration:Bool,
+	?Options:TweenOptions
+}
+
+typedef CubicMotionParams = {
+	Object:FlxObject,
+	FromX:Float,
+	FromY:Float,
+	aX:Float,
+	aY:Float,
+	bX:Float,
+	bY:Float,
+	ToX:Float,
+	ToY:Float,
+	?Duration:Float,
+	?Options:TweenOptions
+}
+
+typedef CircularMotionParams = {
+	Object:FlxObject,
+	CenterX:Float,
+	CenterY:Float,
+	Radius:Float,
+	Angle:Float,
+	Clockwise:Bool,
+	?DurationOrSpeed:Float,
+	?UseDuration:Bool,
+	?Options:TweenOptions
+}
+
+typedef LinearPathOptions = {
+	Object:FlxObject,
+	Points:Array<FlxPoint>,
+	?DurationOrSpeed:Float,
+	?UseDuration:Bool,
+	?Options:TweenOptions
+}
+
+typedef QuadPathParams = {
+	Object:FlxObject,
+	Points:Array<FlxPoint>,
+	?DurationOrSpeed:Float,
+	?UseDuration:Bool,
+	?Options:TweenOptions
+}
+
+typedef TweenParams = OneOfNine<VarTweenParams,AngleTweenParams,ColorTweenParams,NumTweenParams,CircularMotionParams,CubicMotionParams,LinearMotionParams,QuadMotionParams,QuadPathParams>;
+
+enum TweenType
```

Comment:
Calling this "tween type" conflicts with `FlxTween.PINGPONG` which are also considered tween types. It could see that causing quite a bit of confusion.